File: CppPractice/CppLeetcode/Leetcode231.cpp
#include<set>
using namespace std;
class Solution {
public:
    bool isPowerOfTwo(int n) 
    {
        set<int> theSet;
        int kernal = 1;
        theSet.insert(kernal);
        for (int i = 0; i < 32; i++)
        {
            kernal *=2;
            theSet.insert(kernal);
        }

        return theSet.find(n)!=theSet.end();
        
        
    }
};

File: Pythonleetcode/Leetcode231.py
class Solution(object):
    def isPowerOfTwo(self, n):
        if n<=0:
            return False
        s=''
        while n!=0:
            s=str(n%2)+s
            n=n//2
        if '1' not in s[1:]:
            return True
        else: return  False